typescript/packages/plugins/ionic/src/ionic.service.ts

```diff
@@ -0,0 +1,169 @@
+import { Tool } from "@goat-sdk/core";
+import { EVMTransactionOptions, EVMWalletClient } from "@goat-sdk/wallet-evm";
+import { z } from "zod";
+import { parseUnits, formatUnits, type Address } from "viem";
+import type { HealthMetrics, PoolData } from "./types";
+import { ionicProtocolAddresses } from "./ionic.config";
+
+import { poolLensAbi } from "./abis/PoolLens";
+import { comptrollerAbi } from "./abis/Comptroller"; 
+import { poolAbi } from "./abis/Pool";
+import { poolDirectoryAbi } from "./abis/PoolDirectory";
+
+import {
+  SupplyAssetParameters,
+  BorrowAssetParameters,
+  GetHealthMetricsParameters
+} from "./parameters";
+
+export class IonicService {
+  private supportedTokens: string[];
+
+  constructor(supportedTokens?: string[]) {
+    this.supportedTokens = supportedTokens || ["USDC", "WETH"];
+  }
+
+  private async getAssetConfig(chainId: number, symbol: string): Promise<{ address: Address, decimals: number }> {
```

Comment:
I would not implement this here.

This can come from the erc20 plugin. The inputs for each method should be the token address and the amount in base units.

If the prompt is "Supply 3 USDC to pool id 1 in Ionic" the LLM should:
1. Call get token info
2. Call convert to base units
3. Call supply asset with the token address and the amount in base units